app/src/main/java/com/android/periodpals/ui/map/Map.kt

```diff
@@ -85,6 +116,78 @@ fun MapScreen(
   val myLocationOverlay = remember { FolderOverlay() }
   val alertOverlay = remember { FolderOverlay() }
 
+  // To manage the bottom sheet state
+  val sheetState = rememberModalBottomSheetState()
+  val scope = rememberCoroutineScope()
+  var showBottomSheet by remember { mutableStateOf(false) }
+  var content by remember { mutableStateOf(CONTENT.MY_ALERT) }
+  val onMyAlertClick: (Alert) -> Unit = remember {
+    { alert ->
+      showBottomSheet = true
+      content = CONTENT.MY_ALERT
+      alertViewModel.selectAlert(alert)
+    }
+  }
+
+  val onPalAlertClick: (Alert) -> Unit = remember {
+    { alert ->
+      showBottomSheet = true
+      content = CONTENT.PAL_ALERT
+      alertViewModel.selectAlert(alert)
+    }
+  }
+
+  // To manage the alert filter state
+  var isFilterApplied by remember { mutableStateOf(false) }
+  var showFilterDialog by remember { mutableStateOf(false) }
+  var selectedLocation by remember { mutableStateOf<Location?>(null) }
+  var radiusInMeters by remember { mutableDoubleStateOf(DEFAULT_RADIUS) }
+  var productFilter by remember { mutableStateOf<Product?>(Product.NO_PREFERENCE) }
+  var urgencyFilter by remember { mutableStateOf<Urgency?>(null) }
```

Comment:
- Good.
- ```suggestion
    // To manage the alert filter state
    var isFilterApplied by remember { mutableStateOf(false) }
    var showFilterDialog by remember { mutableStateOf(false) }
    var selectedLocation by remember { mutableStateOf<Location?>(null) }
    var radiusInMeters by remember { mutableDoubleStateOf(DEFAULT_RADIUS) }
    var productFilter by remember { mutableStateOf<Product?>(null) }
    var urgencyFilter by remember { mutableStateOf<Urgency?>(null) }
  ```
  Same in `AlertLists`.

Reply:
Oki.

Reply:
Actually, if we want to be consistent with the `AlertLists`, we should leave it like it was:

```kotlin
var productFilter by remember { mutableStateOf<Product?>(Product.NO_PREFERENCE) }
```

I think @agonzalez-r gave me a reason for it too, but I don't remember. It has to do with the way the filter works.

Reply:
Okay if you want, why not.